routes/v0/cases.js

```diff
@@ -1,38 +1,78 @@
-const { Router } = require('express');
 const models = require('../../models');
+const { SafeRouter } = require('../../permissions');
 
 const { Contact, Case, CaseAudit, sequelize } = models;
 const ContactController = require('../../controllers/contact-controller')(Contact);
 const CaseController = require('../../controllers/case-controller')(Case, sequelize);
 const CaseAuditController = require('../../controllers/case-audit-controller')(CaseAudit);
+const { can } = require('../../permissions');
+const { asyncHandler } = require('../../utils');
 
-const casesRouter = Router();
+/**
+ * This middleware checks if the user can edit the case.
+ * If yes, it sets the req.authorized to true.
+ * @param {*} req
+ * @param {*} res
+ * @param {*} next
+ */
+const creatorCanEditCase = async (req, res, next) => {
+  if (!req.authorized) {
+    const { id } = req.params;
+    const caseObj = await CaseController.getCase(id);
+    const canEdit = can(req.user, 'edit', caseObj);
 
-casesRouter.get('/', async (req, res) => {
+    if (canEdit) {
+      req.authorized = true;
+    }
+  }
+
+  next();
+};
```

Comment:
having the code inside `if (!req.authorized)` is more of an optimization than something required by SafeRouter. Imagine an endpoint with 4 different authorization middlewares, if one authorizes it, we don't need to go into the logic of the other ones (especially the ones with DB calls). There are many other ways to achieve this, though.